### ISSUE TITLE
Sync: Failed to rmdir on local storage

### ISSUE THREAD

**Issue by agncr** [rclone/rclone#3878]
#### What is the problem you are having with rclone?
While using sync, rclone won't delete folder with files in it. 

#### What is your rclone version (output from `rclone version`)
rclone v1.50.2

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Debian 10 64bit

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`/usr/bin/rclone -vv --log-file=$log --filter-from=$filter --size-only sync drive:remote /mnt/lokal`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
2020/01/14 03:06:10 DEBUG : Spider-Man - Far From Home (2019): Removing directory
2020/01/14 03:06:10 DEBUG : Spider-Man - Far From Home (2019): Failed to Rmdir: remove /mnt/filme/Spider-Man - Far From Home (2019): directory not empty                                                                   
2020/01/14 03:06:10 DEBUG : Good Boys (2019): Removing directory
2020/01/14 03:06:10 DEBUG : Good Boys (2019): Failed to Rmdir: remove /mnt/filme/Good Boys (2019): directory not empty                                                                                                     
2020/01/14 03:06:10 DEBUG : @Recycle: Removing directory
2020/01/14 03:06:10 DEBUG : @Recycle: Failed to Rmdir: remove /mnt/filme/@Recycle: directory not empty
2020/01/14 03:06:10 DEBUG : Local file system at /mnt/filme: failed to delete 3 directories
2020/01/14 03:06:10 INFO  :

**Comment by ncw** [comment]
Those aren't errors, they are merely debugs.  You are running a filter so rclone doesn't know whether to delete the directories or not so it has a go expecting a failure if there is stuff in them.

Did the sync fail?

**Comment by agncr** [comment]
Hmm, that's true: the sync didn't fail. I think i had a false assumption on how sync works.
Is it possible to sync with a filter and let those files/folders delete locally, which aren't available on the remote any longer?

**Comment by ncw** [comment]
Sync only ever deletes files on the destination...

Can you explain why you want to do that then maybe I can come up with a better workflow?  I'm having trouble figuring out why you'd want to sync files from src -> dst (so src has the master copy) but delete files on src?

one thing you can do is use `rclone lsf` to make lists of files

    rclone lsf -R --files-only src: | sort > src-files
    rclone lsf -R --files-only dst: | sort > dst-files

You can then use the `comm` utility, eg to find all files in the src but not on the dst

    comm -1 src-files dst-files > src-only

Then do something with that list of files, maybe

    rclone delete --dry-run --files-from src-only src:

**Comment by agncr** [comment]
I don't want to delete files from src (remote), instead it should delete from dst (local).

My use case:
I'm using Google Drive as my main media storage. So everything is there and i also sort stuff on that remote. I want to only sync files/folders for the last 2 years to my local storage and everything older is on remote. When i delete files/folders from my remote i want them to be also deleted on local (while syncing).

I could most probably achieve something with lsf, but i thought there would be a "cleaner" solution.

Br

**Comment by ncw** [comment]
Ah I see, I got the direction of your sync wrong - sorry!

What you want is something like this:

    rclone sync remote:path /local/path --dry-run --delete-excluded --max-age 2y

This will sync files from remote: to /local/path. It will only include files that are at most 2 years old. Any files that are deleted on the remote will be deleted on the local.  The `--delete-excluded` means that any files that are more than 2 years old on local will also be deleted.

`--delete-excluded` is a very dangerous flag so do try with `--dry-run` first!

**Comment by agncr** [comment]
I can't use --max-age as i have media on my remote, that's only a few weeks old, but the year is (eg) pre 2000. So i have to use filtering based on the name of the folder. I just tested your suggestion with dry-run: it would also delete files, it shouldn't delete.

`2020/01/17 07:37:15 NOTICE: Ready Player One (2018)/disc.png: Not deleting as --dry-run`

Eventhough my filter looks like this:
`+ **/*(2018)**`
`+ **/*(2019)**`
`- **`

I checked the md5sum of both files (local and remote) and they're identical.
